boot/mbed/include/mcuboot_config/mcuboot_config.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2018 Open Source Foundries Limited
+ * Copyright (c) 2019-2020 Arm Limited
+ * Copyright (c) 2019-2020 Linaro Limited
+ * Copyright (c) 2020 Embedded Planet
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCUBOOT_CONFIG_H__
+#define __MCUBOOT_CONFIG_H__
+
+/*
+ * Signature types
+ *
+ * You must choose exactly one signature type.
+ */
+#define SIGNATURE_TYPE_RSA      0
+#define SIGNATURE_TYPE_EC256    1
+#define SIGNATURE_TYPE_ED25519  2
+
+/* Default to RSA-2048 signatures */
+#if !defined(MCUBOOT_SIGNATURE_ALGORITHM)
+#define MCUBOOT_SIGNATURE_ALGORITHM 0
+#define MCUBOOT_RSA_SIGNATURE_LENGTH 2048
+#endif
+
+#if (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_RSA)
+#define MCUBOOT_SIGN_RSA
+#  if (MCUBOOT_RSA_SIGNATURE_LENGTH != 2048 && \
+       MCUBOOT_RSA_SIGNATURE_LENGTH != 3072)
+#    error "Invalid RSA key size (must be 2048 or 3072)"
+#  else
+#    define MCUBOOT_SIGN_RSA_LEN MCUBOOT_RSA_SIGNATURE_LENGTH
+#  endif
+#elif (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_EC256)
+#define MCUBOOT_SIGN_EC256
+#elif (MCUBOOT_SIGNATURE_ALGORITHM == SIGNATURE_TYPE_ED25519)
+#define MCUBOOT_SIGN_ED25519
+#endif
+
+// TODO - Support for Cryptocell310?
+
+/*
+ * Crypto backend
+ */
+#if (MCUBOOT_CRYPTO_BACKEND == MBEDTLS)
+#define MCUBOOT_USE_MBED_TLS
+#elif (MCUBOOT_CRYPTO_BACKEND == TINYCRYPT)
+#define MCUBOOT_USE_TINYCRYPT
```

Comment:
Since tinycrypt is not currently supported, should we issue a compilation error?

Reply:
I'll add an error for this one, but most other configs that are unsupported/untested do not have a check currently. Will add a check for this one in particular for now.